app/src/main/java/com/android/feedme/MainActivity.kt

```diff
@@ -43,6 +45,7 @@ class MainActivity : ComponentActivity() {
               // Navigation host for the app
               val navController = rememberNavController()
               val navigationActions = NavigationActions(navController)
+              val profileViewModel = ProfileViewModel()
```

Comment:
A better way to instantiate the profileViewModel would be as follows:
`val profileViewModel = viewModel<ProfileViewModel>()`

Let me know what you think, I remember doing it the current way in the bootcamp led to some bugs.

Reply:
I do not understand why. How is this better ? Why do want to declare the type explicitly ? @ohamz

Reply:
I can't explain it well, but this is the response to ChatGPT when I asked which is better : 



- *Using ProfileViewModel() directly instantiates the ViewModel. This approach is straightforward but comes with certain limitations:*

*- Lifecycle Awareness:* The ViewModel is not lifecycle-aware. It won't survive configuration changes like screen rotations. If the activity or fragment is recreated, a new instance of the ViewModel is created, which can lead to loss of data.
*- Resource Management:* It does not handle resources or ongoing processes well. If the ViewModel is managing a process or network call, those could be abruptly stopped or restarted, leading to inefficiencies or errors.


- *Using viewModel<NewViewModel>(), typically seen in Android development with Jetpack's ViewModel architecture, has several advantages:* 

*- Lifecycle Management:* The ViewModel is lifecycle-aware. It survives configuration changes such as screen rotations. This means data isn't lost when the activity or fragment is recreated.
*- Resource Efficiency:* Handles resources more efficiently. Ongoing processes managed by the ViewModel are not interrupted unnecessarily.
*- Scoped to Lifecycle:* It is scoped to the lifecycle of the activity or fragment, ensuring it is cleaned up when the lifecycle ends, preventing memory leaks.

Reply:
Okay, but I dont think the profileViewModel will have life Cycle at all.
The user may frequently check their profile or other person, hence to re-create one is costly.
So I create the instance in mainActivity as never end so it survive configuration.
I don't think the Resource Management make any sense too since it does not die, there should not be any interruption.